### ISSUE TITLE
Commitment saving field not displaying correctly in Power BI and Raw data

### ISSUE THREAD

**Issue by pankaj392** [microsoft/finops-toolkit#604]
Hello Team,

I just installed FinOps hub 2.0 toolkit and uploaded PowerBI cost summary report. While seeing the report the Field commitment saving does not show properly. It has different format as shown in below image.

<img width="396" alt="Issue" src="https://github.com/microsoft/finops-toolkit/assets/140919130/a2d91335-ad9d-4b99-984a-c22e40acfdf3">

Also while checking in Raw data for x_commitmentsaving columns,many fields are filled in with this format. Some rows shows correct format and but some are with below format.

<img width="293" alt="image" src="https://github.com/microsoft/finops-toolkit/assets/140919130/929f94e0-a4af-426d-995d-6f71f69ed84b">

This column format is maintained as below:

<img width="533" alt="image" src="https://github.com/microsoft/finops-toolkit/assets/140919130/404bf12e-5bfa-496f-824d-148de0a2af8e">

Can you please suggest how can I fix this bug or issue?

```[tasklist]
### Tasks
```

**Comment by psilantropy** [comment]
Same problem here. Some more information in my comment here if it helps - https://github.com/microsoft/finops-toolkit/issues/585#issuecomment-1939364005

**Comment by flanakin** [comment]
Turns out there's a bug in Cost Management exports where the x_OnDemandCost is not being set correctly, which is causing the commitment savings calculation to be wrong. They're aware of the issue and have a fix in progress. I'll keep this open until that gets resolved.

While investigating this, I also discovered some other issues impacting savings calculation:

1. List price/cost is missing for some EA rows, which means negotiated discount cost savings is incomplete for EA. (Unrelated to the above issue.)
2. On-demand price is missing for some EA rows, which means commitment discount unit price savings is not available for all records for EA. (Unrelated to the above issue.)
3. On-demand price/cost is missing for all MCA rows, which means we cannot calculate commitment discount savings for MCA. (This was known. Covering it for completeness.)

I'll add some of this into the Commitment discounts report to make it easier to identify these issues.

**Comment by psilantropy** [comment]
> Turns out there's a bug in Cost Management exports

Good to hear. Thanks :). I attempted a few different things it just wasn't working out. Keen to see how our Savings Plan is going. Quite limited reporting information on them.

**Comment by erose96** [comment]
Seeing a similar behavior with my data. Having trouble getting accurate data on Savings Plan savings and this seems like it could solve the issue if fixed.

**Comment by prabha11-sys** [comment]
Any tentative dates on when this will be fixed ?

**Comment by flanakin** [comment]
x_OnDemandCost is in progress. I have a meeting this week to confirm the ETA. I'll see what I can find out regarding the others as well.

**Comment by psilantropy** [comment]
Hi :) If you want some additional input for prioritisation;

I ordered a $50,000 savings plan (1y) in 2023 for the first time. The Savings Plan product on the surface is easy to understand, but I am not given the tools to figure out if I am running it at a loss and it almost feels intentional.

Our Savings plan was 100% for a few months, then has been bouncing everywhere 50-100% after some shutdowns over Christmas holidays.

I did use [this guide](https://learn.microsoft.com/en-us/azure/cost-management-billing/savings-plan/calculate-ea-savings-plan-savings) - but that is next-level annoying (the end result didn't look right either).

It was very easy for me to justify this purchase to the business, but if I was asked to explain the benefit to date from this 50k, it's seemingly impossible. I am doubtful I will renew this.
